Build with Fujitsu Compiler

I use Spack to evaluate Fujitsu Compiler for Fugaku and FX1000.
When I tried to build this package(v1.7.0), I received an error similar to the following:
```
==> Error: ProcessError: Command exited with status 1:
    '/fefs/spack2/spack/opt/spack/linux-rhel8-a64fx/fj-4.10.0/cmake-3.27.4-o4w4ad5gemgpf5teeimtmxl3mpktqzom/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/fefs/spack2/spack/opt/spack/linux-rhel8-a64fx/fj-4.10.0/gftl-shared-1.7.0-t4vfydc53scqqqketa5uiqwefcfkezkh' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/fefs/spack2/spack/opt/spack/linux-rhel8-a64fx/fj-4.10.0/gftl-shared-1.7.0-t4vfydc53scqqqketa5uiqwefcfkezkh/lib;/fefs/spack2/spack/opt/spack/linux-rhel8-a64fx/fj-4.10.0/gftl-shared-1.7.0-t4vfydc53scqqqketa5uiqwefcfkezkh/lib64' '-DCMAKE_PREFIX_PATH:STRING=/fefs/spack2/spack/opt/spack/linux-rhel8-a64fx/fj-4.10.0/gftl-1.11.0-rqedzzjschqifkbb3rvn3spanukrbagq;/fefs/spack2/spack/opt/spack/linux-rhel8-a64fx/fj-4.10.0/cmake-3.27.4-o4w4ad5gemgpf5teeimtmxl3mpktqzom' '/fefs/spack2/tmp/spack-stage/spack-stage-gftl-shared-1.7.0-t4vfydc53scqqqketa5uiqwefcfkezkh/spack-src'

1 error found in build log:
     9     
     10    
     11    -- The Fortran compiler identification is Fujitsu 4.10.0
     12    -- Detecting Fortran compiler ABI info
     13    -- Detecting Fortran compiler ABI info - done
     14    -- Check for working Fortran compiler: /fefs/spack2/spack/lib/spack/env/fj/frt - skipped
  >> 15    CMake Error at CMakeLists.txt:35 (include):
     16      include could not find requested file:
     17    
     18        Fujitsu
     19    
     20    
     21    -- Performing Test _LOGICAL_DEFAULT_KIND: SUCCESS (value=4)
```

I think it is because there is no `Fujitsu.cmake` in [cmake](https://github.com/Goddard-Fortran-Ecosystem/gFTL-shared/tree/main/cmake) directory.
Can you add the file to the cmake directory if I suggest the necessary options?
If it's not easy, I will post a patch to Spack.

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -14,23 +14,13 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-20.04, ubuntu-22.04, macos-11, macos-12]
-        compiler: [gfortran-9, gfortran-10, gfortran-11, gfortran-12]
+        os: [ubuntu-22.04, macos-11, macos-12]
+        compiler: [gfortran-10, gfortran-11, gfortran-12, gfortran-13]
         exclude:
-          - os: ubuntu-20.04
-            compiler: gfortran-11
-          - os: ubuntu-20.04
-            compiler: gfortran-12
-          - os: ubuntu-22.04
-            compiler: gfortran-9
           - os: macos-11
-            compiler: gfortran-12
-          - os: macos-12
-            compiler: gfortran-9
+            compiler: gfortran-13
           - os: macos-12
             compiler: gfortran-10
-          - os: macos-12
-            compiler: gfortran-12
 
       # fail-fast if set to 'true' here is good for production, but when
       # debugging, set to 'false'. fail-fast means if *any* ci test in the matrix fails
@@ -51,36 +41,120 @@ jobs:
 
     name: ${{ matrix.os }} / ${{ matrix.compiler }}
     steps:
-      - name: Install GCC 12 on Ubuntu 22.04
-        if: matrix.os == 'ubuntu-22.04' && matrix.compiler == 'gfortran-12'
-        run: |
-          sudo apt-get install gfortran-12 -y
       - name: Compiler Versions
         run: |
           ${FC} --version
           cmake --version
+
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
         with:
           fetch-depth: 1
+
+      - name: Set all directories as git safe
+        run: |
+          git config --global --add safe.directory '*'
+
       - name: Build gFTL-shared
         run: |
           mkdir -p build
           cd build
           cmake .. -DCMAKE_Fortran_COMPILER=${FC}
           make -j$(nproc)
+
+      - name: Build Tests
+        run: |
+          cd build
+          make -j$(nproc) tests
+
+      - name: Run Tests
+        run: |
+          cd build
+          ctest -j1 --output-on-failure --repeat until-pass:4
+
       - name: Archive log files on failure
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@v4
         if: failure()
         with:
           name: logfiles
           path: |
             build/**/*.log
 
+  Intel:
+    runs-on: ubuntu-20.04
+
+    env:
+      FC: ifx
+      CC: icx
+
+    name: Intel Fortran
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+
+      - name: Set all directories as git safe
+        run: |
+          git config --global --add safe.directory '*'
+
+      - name: Setup Intel oneAPI repository
+        run: |
+          wget https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
+          sudo apt-key add GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
+          sudo add-apt-repository "deb https://apt.repos.intel.com/oneapi all main"
+          sudo apt-get update
+
+      - name: Install Intel oneAPI compilers
+        timeout-minutes: 5
+        run: sudo apt-get install intel-oneapi-compiler-fortran intel-oneapi-compiler-dpcpp-cpp
+
+      # optional
+      - name: Install Intel MPI
+        timeout-minutes: 5
+        run: sudo apt-get install intel-oneapi-mpi intel-oneapi-mpi-devel
+
+      - name: Setup Intel oneAPI environment
+        run: |
+          source /opt/intel/oneapi/setvars.sh
+          printenv >> $GITHUB_ENV
+          printenv | grep intel
+
+      - name: Versions
+        run: |
+          ${FC} --version
+          ${CC} --version
+          mpirun --version
+          cmake --version
+
+      - name: Build gFTL-shared
+        run: |
+          mkdir -p build
+          cd build
+          cmake .. -DCMAKE_Fortran_COMPILER=${FC}
+          make -j$(nproc)
+
+      - name: Build Tests
+        run: |
+          cd build
+          make -j$(nproc) tests
+
+      - name: Run Tests
+        run: |
+          cd build
+          ctest -j1 --output-on-failure --repeat until-pass:4
+
+      - name: Archive log files on failure
+        uses: actions/upload-artifact@v4
+        if: failure()
+        with:
+          name: logfiles
+          path: |
+            build/**/*.log
 
   Nvidia:
     runs-on: ubuntu-20.04
-    container: nvcr.io/nvidia/nvhpc:20.11-devel-cuda11.1-ubuntu20.04
+    container: nvcr.io/nvidia/nvhpc:24.1-devel-cuda12.3-ubuntu20.04
     env:
       FC: nvfortran
 
@@ -90,13 +164,38 @@ jobs:
         run: |
           ${FC} --version
           cmake --version
+
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
         with:
           fetch-depth: 1
+
+      - name: Set all directories as git safe
+        run: |
+          git config --global --add safe.directory '*'
+
       - name: Build gFTL-shared
         run: |
           mkdir -p build
           cd build
           cmake .. -DCMAKE_Fortran_COMPILER=${FC}
           make -j$(nproc)
+
+      - name: Build Tests
+        run: |
+          cd build
+          make -j$(nproc) tests
+
+      - name: Run Tests
+        run: |
+          cd build
+          ctest -j1 --output-on-failure --repeat until-pass:4
+
+      - name: Archive log files on failure
+        uses: actions/upload-artifact@v4
+        if: failure()
+        with:
+          name: logfiles
+          path: |
+            build/**/*.log
+

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -154,7 +154,7 @@ jobs:
 
   Nvidia:
     runs-on: ubuntu-20.04
-    container: nvcr.io/nvidia/nvhpc:24.1-devel-cuda12.3-ubuntu20.04
+    container: nvcr.io/nvidia/nvhpc:24.1-devel-cuda12.3-ubuntu22.04
     env:
       FC: nvfortran
 

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -174,6 +174,11 @@ jobs:
         run: |
           git config --global --add safe.directory '*'
 
+      - name: Add python-is-python3 package
+        run: |
+          apt-get update
+          apt-get install -y python-is-python3
+
       - name: Build gFTL-shared
         run: |
           mkdir -p build

--- a/ChangeLog.md
+++ b/ChangeLog.md
@@ -5,8 +5,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
-## [1.7.0] - 2023-11-29
+### Added
 
+- Fujitsu compiler support
+
+## [1.7.0] - 2023-11-29
 
 ### Fixed
 

--- a/cmake/Fujitsu.cmake
+++ b/cmake/Fujitsu.cmake
@@ -1,0 +1,14 @@
+if (CMAKE_Fortran_COMPILER_ID MATCHES Fujitsu)
+    if (CMAKE_Fortran_COMPILER_VERSION VERSION_LESS 4.10.0)
+      message(FATAL_ERROR "${CMAKE_Fortran_COMPILER_ID} version must be at least 4.10.0!")
+    endif ()
+endif ()
+
+# Compiler specific flags for Fujitsu Fortran compiler
+
+set(check_all "-Nquickdbg")
+set(cpp "-Cfpp")
+
+set(CMAKE_Fortran_FLAGS_DEBUG "-O0 ${check_all}")
+set(CMAKE_Fortran_FLAGS_RELEASE "-O3")
+set(CMAKE_Fortran_FLAGS "-g ${cpp} -Nalloc_assign -Free")

